value_test.go

```diff
@@ -588,32 +633,105 @@ func TestValue_InList(t *testing.T) {
 		Data: []int{1, 2, 3, 4},
 	}
 
-	NewValue(reporter, data1).InList().chain.assert(t, failure)
-	NewValue(reporter, data2).NotInList().chain.assert(t, failure)
-
-	NewValue(reporter, data1).InList(data1, data3).chain.assert(t, success)
-	NewValue(reporter, data2).NotInList(data1, data3).chain.assert(t, success)
-
-	NewValue(reporter, data1).InList(data2, data3).chain.assert(t, failure)
-	NewValue(reporter, data2).NotInList(data2, data3).chain.assert(t, failure)
-
-	NewValue(reporter, data1).InList(data2).chain.assert(t, failure)
-	NewValue(reporter, data2).NotInList(data2).chain.assert(t, failure)
-
-	NewValue(reporter, data1).InList(data1).chain.assert(t, success)
-	NewValue(reporter, data2).NotInList(data1).chain.assert(t, success)
-
-	NewValue(reporter, nil).InList(map[string]interface{}(nil)).chain.assert(t, success)
-	NewValue(reporter, nil).NotInList(map[string]interface{}{}).chain.assert(t, success)
+	t.Run("basic", func(t *testing.T) {
+		cases := []struct {
+			name          string
+			value         interface{}
+			list          []interface{}
+			wantInList    chainResult
+			wantNotInList chainResult
+		}{
+			{
+				name:          "no list",
+				value:         data1,
+				list:          nil,
+				wantInList:    failure,
+				wantNotInList: failure,
+			},
+			{
+				name:          "empty list",
+				value:         data1,
+				list:          []interface{}{},
```

Comment:
While I was doing this issue, I noticed that this is not the same as NewValue(reporter, data1).InList([]interface{}{}).
Because in tests it's an empty list, but if you do it as it was originally, just in-line, it will be a list containing a slice of interfaces.
I've seen it in `TestNumber_InList/basic`

Reply:
Good catch, let's keep the new version.

I think both "empty list" and "no list" should be moved to "invalid argument" subtest, because they both are reported as usage error.

Also let's rename "no list" to "nil list".